src/main/java/producer/Producer.java

```diff
@@ -0,0 +1,6 @@
+package producer;
+
+public interface Producer<K, V> {
+    boolean send(ProducerRecord<K,V> record);
```

Comment:
Out of curiosity, what does the boolean represent in the context of this method? What would true/false mean?

Reply:
true = record successfully sent
false = record not successfully sent

Reply:
Ah got it, sent successfully to where?

Reply:
so the idea is all records will be sent to a buffer (`RecordAccumulator`) to hold records for batching purposes and ig it'll return true or not if it was successfully stored in said buffer, but now that i'm thinking of it may not entirely be 100% necessary to return a bool for the process. Will have to ponder

Reply:
Yea im thinking it can be void

Reply:
Non-blocking though